File: uniqueChar.cpp
#include<bits/stdc++.h>
using namespace std;

bool isUnique(string str){
    transform(str.begin(), str.end(), str.begin(), ::tolower);
    if (str.length() > 128){
        return false;
    }
    bool charSet[128] = {false};
    for (int i = 0; i < str.length(); i++)
    {
        int val = str[i];
        if (charSet[val])
        {
            return false;
        }
        charSet[val] = true;
    }
    return true;
}

int main(){
    string str = "Aman";
    bool ans = isUnique(str);
    cout << boolalpha << ans << endl;
    return 0;
}

File: unique_char.py
def isUnique(str):
    str = str.lower()
    if len(str) > 128:
        return False
    char_set = [False] * 128
    for i in range(len(str)):
        val = ord(str[i])
        if char_set[val]:
            return False
        char_set[val] = True
    return True

str = "Aman"
print(isUnique(str))